### PR TITLE
initial module module implementation

### PR DESCRIPTION
**Pull request by mcalhoun**
## what
* initial implementation

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -70,15 +81,14 @@ usage: |-
 
   ```hcl
   module "example" {
-    source = "https://github.com/cloudposse/terraform-example-module.git?ref=master"
-    example = "Hello world!"
+    source = "https://github.com/cloudposse/terraform-aws-guardduty.git?ref=master"
```
Comment (osterman) [inline README.yaml:84]: Shouldn't we provide an event rule example?

https://summitroute.com/blog/2019/03/06/guardduty_event_collection_via_cloudwatch_events/

File: main.tf [hunk 2]
```diff
@@ -1,13 +1,89 @@
...
+    principals {
+      type        = "Service"
+      identifiers = ["cloudwatch.amazonaws.com"]
+    }
+    resources = [module.sns_topic[0].sns_topic.arn]
+    effect    = "Allow"
   }
 }
 
+resource "aws_cloudwatch_event_rule" "findings" {
```
Comment (osterman) [inline main.tf:58]: @aknysh @mcalhoun does it make sense to have a `catalog/` of event rules similar to we do for SCP/datadog/etc?

Comment (aknysh) [reply main.tf:58]: if those rules can be configured in YAML, then yes, `catalog` would be nice and would follow the same pattern we use in:

https://github.com/cloudposse/terraform-aws-service-control-policies/tree/master/catalog

https://github.com/cloudposse/terraform-datadog-monitor/tree/master/catalog

We should use `terraform-yaml-config` to read the from YAML catalog:
https://github.com/cloudposse/terraform-aws-service-control-policies/blob/master/examples/complete/main.tf#L2

File: README.yaml (hunk 1 above)
Comment (mcalhoun) [reply README.yaml:84]: 647180542b4378772546e7e01ebb0557d941eab6

File: main.tf (hunk 2 above)
Comment (mcalhoun) [reply main.tf:58]: There aren't really any configurable event rules for Guard Duty, the only config that would be applicable to this comment would be a catalog of SNS subscriptions, but probably doesn't make sense as the variability in params would make it difficult.

File: README.yaml [hunk 3]
```diff
@@ -37,28 +37,39 @@ badges:
 # List any related terraform modules that this module may be used with or that this module depends on.
 related:
   - name: "terraform-null-label"
-    description: "Terraform module designed to generate consistent names and tags for resources. Use terraform-null-label to implement a strict naming convention."
+    description:
```
Comment (aknysh) [inline README.yaml:40]: add more related?
all config/security modules?

File: examples/complete/outputs.tf [hunk 4]
```diff
@@ -1,14 +1,3 @@
...
-  description = "Output \"example\" from example module"
-  value       = module.example.example
-}
-
-output "random" {
-  description = "Output \"random\" from example module"
-  value       = module.example.random
+output "sns_topic" {
+  value = module.example.sns_topic
 }
```
Comment (aknysh) [inline examples/complete/outputs.tf:3]: add other outputs?, e.g. from

resource "aws_guardduty_detector" "guardduty"

File: main.tf [hunk 5]
```diff
@@ -1,13 +1,89 @@
...
+  sqs_dlq_enabled = false
+
+  context = module.this.context
+}
+
+module "findings_label" {
+  source  = "cloudposse/label/null"
+  version = "0.21.0"
+
+  attributes = ["guardduty-findings"]
```
Comment (aknysh) [inline main.tf:31]:
```suggestion
  attributes = concat(module.this.attributes, ["guardduty-findings"])
```

File: main.tf [hunk 6]
```diff
@@ -1,13 +1,89 @@
...
+  count     = module.this.enabled && local.create_sns_topic ? 1 : 0
+  policy_id = "GuardDutyPublishToSNS"
+  statement {
+    sid = ""
+    actions = [
+      "sns:Publish"
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["cloudwatch.amazonaws.com"]
```
Comment (aknysh) [inline main.tf:51]: not sure, but is this a correct identifier? ["cloudwatch.amazonaws.com"]

**Review by aknysh — COMMENTED** [review]
LGTM, just a few nitpicks

File: main.tf (hunk 5 above)
Comment (mcalhoun) [reply main.tf]: e25b0b463fb586bd3ff991ce438238e21ad12f39

File: main.tf (hunk 6 above)
Comment (mcalhoun) [reply main.tf:51]: I also thought it should be `events.amazonaws.com`, but after applying it was switched to `cloudwatch.amazonaws.com`, so there was a perpetual diff. I tested with the identifier set to `cloudwatch.amazonaws.com` and it worked (sent notifications to SNS).

File: examples/complete/outputs.tf (hunk 4 above)
Comment (mcalhoun) [reply examples/complete/outputs.tf:3]: b55a797e96519d7ff0396dcf98a175ce1f6c8267

File: README.yaml (hunk 3 above)
Comment (mcalhoun) [reply README.yaml:40]: f22831a2e73029ccd7661840128dc6b57170ad61

File: main.tf [hunk 7]
```diff
@@ -1,13 +1,89 @@
...
+# https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-integration-types.html
+# https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/resource-based-policies-cwe.html#sns-permissions
+#-----------------------------------------------------------------------------------------------------------------------
+module "sns_topic" {
+
+  source  = "cloudposse/sns-topic/aws"
+  version = "0.9.0"
+  count   = local.create_sns_topic ? 1 : 0
+
+  attributes      = ["guardduty"]
```
Comment (aknysh) [inline main.tf:20]:
```suggestion
  attributes      = concat(module.this.attributes, ["guardduty"])
```

Comment (mcalhoun) [reply main.tf]: 540bcae5464c604a3427134721a271f74c1c900e

**Review by aknysh — APPROVED** [review]

**Pull request merged**